### PR TITLE
Qute: implement strategies if a property is not found in an expression

### LINKED ISSUE #12933: Qute: if needed throw an exception if a property is not found
Currently, we output the `NOT_FOUND` constant if a property is not found when evaluating an expression. It should be possible to register a `ResultMapper` that simly throws an exception for `io.quarkus.qute.Results.Result.NOT_FOUND`. We could make it configurable.

### PR DESCRIPTION
**Pull request by mkouba**
- resolves #12933

### CODE REVIEW

File: extensions/qute/runtime/src/main/java/io/quarkus/qute/runtime/EngineProducer.java [hunk 1]
```diff
@@ -76,14 +76,30 @@ public EngineProducer(QuteContext context, Event<EngineBuilder> builderReady, Ev
...
+        // If needed use a specific result mapper for the selected strategy  
+        switch (runtimeConfig.propertyNotFoundStrategy) {
+            case THROW_EXCEPTION:
+                builder.addResultMapper(new PropertyNotFoundThrowException());
+                break;
+            case NOOP:
+                builder.addResultMapper(new PropertyNotFoundNoop());
+                break;
+            case OUTPUT_ORIGINAL:
+                builder.addResultMapper(new PropertyNotFoundOutputExpression());
```
Comment (gastaldi) [inline extensions/qute/runtime/src/main/java/io/quarkus/qute/runtime/EngineProducer.java:88]:
```suggestion
                builder.addResultMapper(new PropertyNotFoundOutputOriginal());
```

File: extensions/qute/runtime/src/main/java/io/quarkus/qute/runtime/PropertyNotFoundOutputExpression.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package io.quarkus.qute.runtime;
+
+import io.quarkus.qute.Expression;
+import io.quarkus.qute.ResultMapper;
+import io.quarkus.qute.Results.Result;
+import io.quarkus.qute.TemplateNode.Origin;
+
+class PropertyNotFoundOutputExpression implements ResultMapper {
```
Comment (gastaldi) [inline extensions/qute/runtime/src/main/java/io/quarkus/qute/runtime/PropertyNotFoundOutputExpression.java:8]:
```suggestion
class PropertyNotFoundOutputOriginal implements ResultMapper {
```

**Review by gastaldi — COMMENTED** [review]
Looks good but I miss some docs for this feature.

**Comment by mkouba** [comment]
> Looks good but I miss some docs for this feature.

Well, it will be listed in the ref guide in the [Configuration Reference](https://quarkus.io/guides/qute-reference#configuration-reference). I don't think it's worth a special section...

**Review by gastaldi — APPROVED** [review]
👍🏻

**Comment by gsmet** [comment]
I wonder if NOT_FOUND is the best default value.

**Comment by mkouba** [comment]
> I wonder if NOT_FOUND is the best default value.

Feel free to propose a new one ;-). However, this PR is not about the default value.

**Comment by gsmet** [comment]
Well, you're adding new strategies so I think choosing the default behavior is a natural thing for this PR.

Personally, I think I would prefer having an error by default because otherwise it's easy to miss but maybe I'm being too aggressive.

**Comment by mkouba** [comment]
> Well, you're adding new strategies so I think choosing the default behavior is a natural thing for this PR.

Yes, we're adding new strategies but we don't change the default one.

> Personally, I think I would prefer having an error by default because otherwise it's easy to miss but maybe I'm being too aggressive.

Well, it's not so clear. It depends on the use case. The good news is that it should not happen if typesafe templates and/or checked expresions are used ;-).

**Pull request merged**